src/components/tamagotchi-new-ui.tsx

```diff
@@ -0,0 +1,320 @@
+"use client";
+/**
+ * This code was generated by v0 by Vercel.
+ * @see https://v0.dev/t/JO1Z5J3ewhA
+ */
+
+import {
+  CardTitle,
+  CardDescription,
+  CardHeader,
+  CardContent,
+  CardFooter,
+  Card,
+} from "@/components/ui/card";
+import { Button } from "@/components/ui/button";
+import {
+  FaBowlFood,
+  FaBasketball,
+  FaBath,
+  FaFaceSadCry,
+  FaChartArea,
+} from "react-icons/fa6";
+import { useEffect, useState } from "react";
+import { death, idle } from "./tamagotchiFrames";
+import { MdLocalHospital } from "react-icons/md";
+import { INTERACTION } from "@/app/utils/interaction";
+
+const DEFAULT_STATUS = ":)";
+export function TamagotchiNewUI() {
+  const [frameIndex, setFrameIndex] = useState<number>(0);
+  const [tamagotchiState, setTamagotchiState] = useState<any>({});
+  const [animation, setAnimation] = useState<string[]>(idle);
+  const [tamaStatus, setTamaStatus] = useState(DEFAULT_STATUS);
+  const [checkingStatus, setCheckingStatus] = useState(false);
+  const [isInteracting, setIsInteracting] = useState(false);
+
+  //TODO - call init endpoint to determine if tamagotchi is initialized. if not generate one.
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch("/api/getState", { method: "POST" });
+        if (response.ok) {
+          const jsonData = await response.json();
+          if (!!!jsonData.comment) {
+            jsonData.comment = "No comments";
+          }
+          setTamagotchiState(jsonData);
+          console.log(jsonData);
+          handleDeath(jsonData, pollInterval);
+        }
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+
+    // Cycle through frames every 1 second
+    const frameInterval = setInterval(() => {
+      setFrameIndex((prevIndex) => (prevIndex + 1) % idle.length);
+    }, 1000);
+
+    // Fetch data on component mount
+    fetchData();
+
+    // Start polling data every N seconds (adjust the interval as needed)
+    const pollInterval = setInterval(fetchData, 5000); // Poll every 5 seconds
+
+    return () => {
+      clearInterval(pollInterval);
+      clearInterval(frameInterval);
+    };
+  }, []);
+
+  const handleResponse = (responseText: string) => {
+    const responseJSON = JSON.parse(responseText);
+    const animation = JSON.parse(responseJSON.animation);
+    const status = responseJSON.status;
+    setFrameIndex(0);
+    setAnimation(animation);
+    setTamaStatus(status);
+  };
+
+  const handleDeath = (jsonData: any, pollInterval: NodeJS.Timer) => {
+    if (jsonData.death) {
+      setAnimation(death);
+      setTamaStatus("Dead :(");
+      clearInterval(pollInterval);
+    }
+  };
+
+  const handleBath = async () => {
```

Comment:
**issue (complexity):** Consider consolidating the duplicate interaction handlers into a single unified handler function.

The interaction handlers contain significant duplicated logic that can be simplified into a single unified handler while maintaining all functionality. Here's how:

```typescript
const INTERACTION_LABELS = {
  [INTERACTION.FEED]: "Feeding",
  [INTERACTION.PLAY]: "Playing",
  [INTERACTION.BATH]: "Bathing",
  [INTERACTION.DISCIPLINE]: "Disciplining",
  [INTERACTION.GO_TO_HOSPITAL]: "Treating",
} as const;

const handleInteraction = async (interactionType: INTERACTION) => {
  setIsInteracting(true);
  setTamaStatus(`${INTERACTION_LABELS[interactionType]}...`);

  try {
    const response = await fetch("/api/interact", {
      method: "POST",
      body: JSON.stringify({ interactionType }),
      headers: { "Content-Type": "application/json" },
    });
    handleResponse(await response.text());
  } catch (e) {
    console.log(e);
  }

  setTimeout(() => {
    setAnimation(idle);
    setTamaStatus(DEFAULT_STATUS);
    setIsInteracting(false);
  }, 9000);
};
```

Then replace all interaction handlers with:

```typescript
const feedTamagotchi = () => handleInteraction(INTERACTION.FEED);
const playWithTamagotchi = () => handleInteraction(INTERACTION.PLAY);
const handleBath = () => handleInteraction(INTERACTION.BATH);
// etc...
```

This reduces code duplication while keeping all functionality intact. It's also more maintainable as changes to interaction handling only need to be made in one place.